src/NetMQ/NetMQSocket.cs

```diff
@@ -400,6 +400,19 @@ public void Monitor([NotNull] string endpoint, SocketEvents events = SocketEvent
             m_socketHandle.Monitor(endpoint, events);
         }
 
+        /// <summary>
+        /// stop monitoring all enpoints for SocketEvent events.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">This object is already disposed.</exception>
+        /// <exception cref="TerminatingException">The socket has been stopped.</exception>
+        /// <exception cref="NetMQException">on failure to stop socket.</exception>
+        public void StopMonitor()
+        {
+            m_socketHandle.CheckDisposed();
+
+            m_socketHandle.Monitor(null, SocketEvents.All);
+        }
+
```

Comment:
I think it would be good to add a PR to the main package as well, so that we can get feedback from NetMQ creators and make use of the official version in the future